src/emucore/Random.hxx

```diff
@@ -54,11 +55,13 @@ class Random
     */
     uInt32 next();
 
-  private:
-    // Indicates the next random number
-    uInt32 myValue;
+  public:
+    static unsigned long max();
```

Comment:
In fact, the method seed() is also static. The unique method that is not
static is next(). I made such method static because it is not dependent of
a particular object, but if you thing I must change this, I'll do it.

On 25 May 2015 at 05:32, Marc G. Bellemare <notifications@github.com> wrote:

> In src/emucore/Random.hxx
> <https://github.com/mgbellemare/Arcade-Learning-Environment/pull/55#discussion_r30975437>
> :
>
> > @@ -54,11 +55,13 @@ class Random
> >      */
> >      uInt32 next();
> >
> > -  private:
> > -    // Indicates the next random number
> > -    uInt32 myValue;
> > +  public:
> > +    static unsigned long max();
>
> Is there a reason for making this static (when the other methods aren't)?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mgbellemare/Arcade-Learning-Environment/pull/55/files#r30975437>
> .
>



-- 
Marlos Cholodovskis Machado
http://www.dcc.ufmg.br/~marlos <http://www.cs.ualberta.ca/~machado>